qiskit/providers/provider.py

```diff
@@ -75,3 +75,108 @@ def __eq__(self, other):
         equal. Subclassed providers can override this behavior.
         """
         return type(self).__name__ == type(other).__name__
+
+
+class ProviderV2(Provider, ABC):
+    """Base class for a Backend Provider."""
+    version = 2
+
+    def get_backend(self, name=None, **kwargs):
+        """Return a single backend matching the specified filtering.
+
+        Args:
+            name (str): name of the backend.
+            **kwargs: dict used for filtering.
+
+        Returns:
+            Backend: a backend matching the filtering.
+
+        Raises:
+            QiskitBackendNotFoundError: if no backend could be found or
+                more than one backend matches the filtering criteria.
+        """
+        backends = self.backends(name, **kwargs)
+        if len(backends) > 1:
+            raise QiskitBackendNotFoundError('More than one backend matches the criteria')
+        if not backends:
+            raise QiskitBackendNotFoundError('No backend matches the criteria')
+
+        return backends[0]
+
+    @abstractmethod
+    @property
+    def backends(self):
```

Comment:
Based on the docstring, shouldn't this be `backend` (without the `s`)? ibmq also just changed from `provider.backends.<backend_name>` to `provider.backend.<backend_name>`.